### PR TITLE
#588 - Pediatric Abdominal CT Segmentation Bundle

### PR DESCRIPTION
**Pull request by EmergentBehaviour**
### Description
This is a submission for the MONAI Bundle of a Pediatric Abdominal CT Segmentation model from the Children's Artificial Intelligence Imaging Research (CAIIR) Center at Cincinnati Children's Hospital.

Metadata check passed, torchdynamo could not install / run during code-formatting script check. Open to any fixes or alternatives for code-formatting check / confirmation.
Please feel free to write any questions, comments or suggestions.

### Status
Ready for Feedback / Publication

### Please ensure all the checkboxes:

- [x] Codeformat tests passed locally by running `./runtests.sh --codeformat`.
- [x] In-line docstrings updated.
- [x] Update `version` and `changelog` in `metadata.json` if changing an existing bundle.
- [x] Please ensure the naming rules in config files meet our requirements (please refer to: `CONTRIBUTING.md`).
- [x] Ensure versions of packages such as `monai`, `pytorch` and `numpy` are correct in `metadata.json`.
- [x] Descriptions should be consistent with the content, such as `eval_metrics` of the provided weights and TorchScript modules.
- [x] Files larger than 25MB are excluded and replaced by providing download links in `large_files.yml`.
- [x] Avoid using path that contains personal information within config files (such as use `/home/your_name/` for `"bundle_root"`).

### CODE REVIEW

**Comment by yiheng-wang-nv** [comment]
Hi @EmergentBehaviour , could you check the failed CI tests? Thanks!

**Comment by EmergentBehaviour** [comment]
> Hi @EmergentBehaviour , could you check the failed CI tests? Thanks!

Hello, I've since fixed the dependency versioning that caused the pre-merge to fail, however running some of the ci tests locally (like that contained within ci/run_regular_tests_cpu.sh) have errors with the checksum to download the latest bundles:

Traceback (most recent call last):
  File "/workspace/model-zoo/ci/download_latest_bundle.py", line 44, in <module>
    download_latest_bundle(bundle_name, models_path, download_path)
  File "/workspace/model-zoo/ci/download_latest_bundle.py", line 23, in download_latest_bundle
    checksum = get_version_checksum(bundle_name=bundle_name, version=version, model_info_path=model_info_path)
  File "/workspace/model-zoo/ci/utils.py", line 128, in get_version_checksum
    return model_info_dict[f"{bundle_name}_v{version}"]["checksum"]
KeyError: '_v1.0.9'

Is there something I can do to fix this?

**Comment by EmergentBehaviour** [comment]
Awaiting review unless there's a way to run blossom on my end.

**Comment by ericspod** [comment]
hi @EmergentBehaviour sorry for the delay. I've run the other tests but can't invoke blossom, there are some things to fix still it seems but then @KumoLiu can trigger the blossom tests. Thanks!

**Comment by yiheng-wang-nv** [comment]
HI @EmergentBehaviour , as for the code format check, the current error is:
```
ERROR: /home/runner/work/model-zoo/model-zoo/models/pediatric_abdominal_ct_segmentation/scripts/compute_metric.py Imports are incorrectly sorted and/or formatted.
```
Could you use `isort` to adjust the import order to fix it? Thanks!

**Comment by yiheng-wang-nv** [comment]
As for the premerge cpu check, since your bundle does not have `model.pt`, could you add the bundle name into:
https://github.com/Project-MONAI/model-zoo/blob/dev/ci/bundle_custom_data.py#L26

You can also see the contributing guide: https://github.com/Project-MONAI/model-zoo/blob/dev/CONTRIBUTING.md#preferred-files-and-keys Thanks! @EmergentBehaviour

**Comment by EmergentBehaviour** [comment]
Thank you for the response, and I have made the according modifications. However, this will still not pass the code format test as when the codebase is formatted to conform with isort, there are issues with the black standard according to the code formatting check, and vice versa, despite having used fixes that should in theory keep both standards compliant.

Let me know if there's anything I can do to ameliorate this or if you've run across this with the code formatting scripts.

File: models/pediatric_abdominal_ct_segmentation/configs/metadata.json [hunk 1]
```diff
@@ -0,0 +1,101 @@
+{
+    "schema": "https://github.com/Project-MONAI/MONAI-extra-test-data/releases/download/0.8.1/meta_schema_20220324.json",
+    "version": "0.4.3",
+    "changelog": {
+        "0.4.3": "add support for TensorRT conversion and inference",
```
Comment (yiheng-wang-nv) [inline models/pediatric_abdominal_ct_segmentation/configs/metadata.json]: Hi @EmergentBehaviour , as a new bundle, I think we should only keep only one version in the changelog, could you modify here?

File: models/pediatric_abdominal_ct_segmentation/docs/README.md [hunk 2]
```diff
@@ -0,0 +1,145 @@
...
+- Cincinnati Children's Liver Spleen CT dataset (275)
+- Cincinnati Children's Pancreas CT dataset (146)
+
+Testing data:
+- Cincinnati Children's Liver-Spleen (57)
+- Cincinnati Children's Pancreas (35)
+- TCIA-Pediatric (74)
+- Total Segmentator (50)
+
+External dataset licenses can be found in accompanying text file. Internal datasets currently not publicly available.
```
Comment (yiheng-wang-nv) [inline models/pediatric_abdominal_ct_segmentation/docs/README.md:31]: could you add more instructions on how to prepare external datasets and train this bundle?

**Comment by yiheng-wang-nv** [comment]
> Thank you for the response, and I have made the according modifications. However, this will still not pass the code format test as when the codebase is formatted to conform with isort, there are issues with the black standard according to the code formatting check, and vice versa, despite having used fixes that should in theory keep both standards compliant.
> 
> Let me know if there's anything I can do to ameliorate this or if you've run across this with the code formatting scripts.

Hi @EmergentBehaviour , I pushed a commit and solves the `isort` and `black` issues. You can also do that locally via running:
```
./runtests.sh --autofix
```

There are still flake8 issues: https://github.com/Project-MONAI/model-zoo/actions/runs/9952935547/job/27495300660?pr=589#step:7:90
like:
```
/home/runner/work/model-zoo/model-zoo/models/pediatric_abdominal_ct_segmentation/scripts/compute_metric.py:39:1: F401 'argparse' imported but unused
/home/runner/work/model-zoo/model-zoo/models/pediatric_abdominal_ct_segmentation/scripts/compute_metric.py:41:1: F401 'glob.glob' imported but unused
/home/runner/work/model-zoo/model-zoo/models/pediatric_abdominal_ct_segmentation/scripts/compute_metric.py:43:1: F401 'nibabel as nib' imported but unused
/home/runner/work/model-zoo/model-zoo/models/pediatric_abdominal_ct_segmentation/scripts/compute_metric.py:44:1: F401 'numpy as np' imported but unused
/home/runner/work/model-zoo/model-zoo/models/pediatric_abdominal_ct_segmentation/scripts/compute_metric.py:47:1: F401 'monai.data.create_test_image_3d' imported but unused
/home/runner/work/model-zoo/model-zoo/models/pediatric_abdominal_ct_segmentation/scripts/compute_metric.py:50:1: F401 'monai.transforms.KeepLargestConnectedComponentd' imported but unused
/home/runner/work/model-zoo/model-zoo/models/pediatric_abdominal_ct_segmentation/scripts/compute_metric.py:50:1: F401 'monai.transforms.ScaleIntensityd' imported but unused
...
```
could you resolve them?

File: models/pediatric_abdominal_ct_segmentation/docs/README.md (hunk 2 above)
Comment (EmergentBehaviour) [reply models/pediatric_abdominal_ct_segmentation/docs/README.md:31]: More detailed instructions added.

File: models/pediatric_abdominal_ct_segmentation/configs/metadata.json (hunk 1 above)
Comment (EmergentBehaviour) [reply models/pediatric_abdominal_ct_segmentation/configs/metadata.json]: changelog shortened

**Comment by EmergentBehaviour** [comment]
Just checking to see if blossom will pass. @yiheng-wang-nv

**Review by yiheng-wang-nv — APPROVED** [review]

**Comment by EmergentBehaviour** [comment]
Would there be a reason why blossom shows as failing in this page but successful on its action page? It looks like it skipped the post-processing step but ran it on the dev branch instead around the same time.

**Comment by yiheng-wang-nv** [comment]
Hi @EmergentBehaviour ,
CI failed when check the .ts file:
```
[2024-09-03T08:27:40.606Z] Traceback (most recent call last):
[2024-09-03T08:27:40.606Z]   File "/home/jenkins/agent/workspace/MONAI-zoo-premerge/ci/verify_bundle.py", line 355, in <module>
[2024-09-03T08:27:40.606Z]     verify(bundle, models_path, mode)
[2024-09-03T08:27:40.606Z]   File "/home/jenkins/agent/workspace/MONAI-zoo-premerge/ci/verify_bundle.py", line 343, in verify
[2024-09-03T08:27:40.606Z]     verify_torchscript(bundle_path, net_id, config_file, model_name, ts_name)
[2024-09-03T08:27:40.606Z]   File "/home/jenkins/agent/workspace/MONAI-zoo-premerge/ci/verify_bundle.py", line 219, in verify_torchscript
[2024-09-03T08:27:40.606Z]     torch.jit.load(ts_model_path)
[2024-09-03T08:27:40.606Z]   File "/home/jenkins/agent/workspace/MONAI-zoo-premerge/model_zoo_venv/lib/python3.10/site-packages/torch/jit/_serialization.py", line 162, in load
[2024-09-03T08:27:40.606Z]     cpp_module = torch._C.import_ir_module(cu, str(f), map_location, _extra_files, _restore_shapes)  # type: ignore[call-arg]
[2024-09-03T08:27:40.606Z] RuntimeError: 
[2024-09-03T08:27:40.606Z] Unknown type name '__torch__.torch.classes.tensorrt.Engine':
[2024-09-03T08:27:40.606Z]   File "code/__torch__/monai/networks/nets/dynunet.py", line 4
[2024-09-03T08:27:40.606Z]   __parameters__ = []
[2024-09-03T08:27:40.606Z]   __buffers__ = []
[2024-09-03T08:27:40.606Z]   __torch___monai_networks_nets_dynunet_DynUNet_trt_engine_0x558a0b5a81a0 : __torch__.torch.classes.tensorrt.Engine
[2024-09-03T08:27:40.606Z]                                                                             ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~ <--- HERE
[2024-09-03T08:27:40.606Z]   def forward(input_0: __torch__.monai.networks.nets.dynunet.DynUNet_trt,
[2024-09-03T08:27:40.606Z]     input_1: Tensor) -> Tensor:
```
I also run `torch.jit.load` locally and met the same issue. May I ask how to use this file?

**Comment by EmergentBehaviour** [comment]
@yiheng-wang-nv 

The TensorRT model should be able to load in as-is or if not can be exported from the .pt file as described in the bundle's README and loaded from that new file. May I ask what version of TensorRT / any particular plugins this test is using?

I can alternatively remove the .ts model from the large files list and leave it as an option for users to export from the .pt if desired, as different users may be using different versions.

**Comment by yiheng-wang-nv** [comment]
> @yiheng-wang-nv
> 
> The TensorRT model should be able to load in as-is or if not can be exported from the .pt file as described in the bundle's README and loaded from that new file. May I ask what version of TensorRT / any particular plugins this test is using?
> 
> I can alternatively remove the .ts model from the large files list and leave it as an option for users to export from the .pt if desired, as different users may be using different versions.

thanks, let me add a minor change to make the check optional

**Review by yiheng-wang-nv — APPROVED** [review]

**Pull request merged**